File: 2020/J4.py
string = input()
sub = input()
subs = []
printno = True
for i in sub:
    subs.append(sub)
    sub += sub[0]
    sub = sub[1:]
for sub in subs:
    if string.find(sub) != -1:
        print("yes")
        printno = False
        break
if printno == True:
    print("no")

File: 2020/J4.cpp
#include <bits/stdc++.h>

using namespace std;

int main()
{
    string full;
    string key;
    int found;
    cin >> full;
    cin >> key;
    int length = key.length();
    for (int i = 0; i < length; i++) {
        found = full.find(key);
        if (found != string::npos) {
            printf("yes");
            break;
        }
        key.push_back(key.front());
        key.erase(0, 1);
    }
    if (found == -1) {
        printf("no");
    }
}
